### PR TITLE
add check user permission endpoint

### PR DESCRIPTION
**Pull request by Akopti8**
Add an endpoint that will check the user's permission to preform an operation at a specific prefix, in a specific bucket.

### CODE REVIEW

File: blobstore/blobhandler.go [hunk 1]
```diff
@@ -305,3 +305,22 @@ func (bh *BlobHandler) PingWithAuth(c echo.Context) error {
...
+
+func (bh *BlobHandler) HandleCheckS3UserPermission(c echo.Context) error {
+	if bh.Config.AuthLevel == 0 {
+		log.Info("Checked user permissions successfully")
+		return c.JSON(http.StatusOK, true)
+	}
+	userEmail := c.QueryParam("user_email")
+	s3Prefix := c.QueryParam("s3_prefix")
+	operation := c.QueryParam("operation")
+	if userEmail == "" && s3Prefix == "" {
```
Comment (ShaneMPutnam) [inline blobstore/blobhandler.go:317]: userEmail, s3Prefix, and operation are all required, right? This should be an "or" statement.

File: main.go [hunk 2]
```diff
@@ -108,6 +108,9 @@ func main() {
 	// e.PUT("/object/cross-bucket/copy", auth.Authorize(bh., writers...))
 	// e.PUT("/prefix/cross-bucket/copy", auth.Authorize(bh., writers...))
 
+	//auth
+	e.GET("/check_user_permission", auth.Authorize(bh.HandleCheckS3UserPermission, allUsers...))
```
Comment (ShaneMPutnam) [inline main.go:112]: Not sure the purpose of this endpoint, but should allUsers really be able to check other users permissions? I would think you would want to limit to admins or something?

Comment (ShaneMPutnam) [reply main.go:112]: Thanks!

File: blobstore/blobhandler.go [hunk 3]
```diff
@@ -305,3 +305,22 @@ func (bh *BlobHandler) PingWithAuth(c echo.Context) error {
 
 	return c.JSON(http.StatusOK, bucketHealth)
 }
+
+func (bh *BlobHandler) HandleCheckS3UserPermission(c echo.Context) error {
+	if bh.Config.AuthLevel == 0 {
```
Comment (ShaneMPutnam) [inline blobstore/blobhandler.go:310]: There is some more logic needed here -- if there is not fine grained access control, that does not mean every user can upload, it just means that we are not limited the upload by bucket. A user could be s3_reader and still not be able to upload.

Comment (Akopti8) [reply blobstore/blobhandler.go:310]: The user will not be able to call this endpoint unless they have s3_limited_writer or s3_writer in their roles (it's a secondary check in the front end after checking their writer roles). So I can make those roles and the admin the only people that can call this endpoint!

**Pull request closed without merging**